src/plugins/dashboard/public/dashboard_actions/add_agg_vis_action.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { EmbeddableApiContext, apiHasType, type HasType } from '@kbn/presentation-publishing';
+import { COMMON_EMBEDDABLE_GROUPING } from '@kbn/embeddable-plugin/public';
+import { Action, IncompatibleActionError } from '@kbn/ui-actions-plugin/public';
+import { DASHBOARD_APP_ID } from '../dashboard_constants';
+import { pluginServices } from '../services/plugin_services';
+
+const ADD_AGG_VIS_ACTION_ID = 'ADD_AGG_VIS';
+
+type AddAggVisualizationPanelActionApi = HasType;
+
+const isApiCompatible = (api: unknown | null): api is AddAggVisualizationPanelActionApi => {
+  return apiHasType(api) && api.type === 'dashboard';
+};
+
+export class AddAggVisualizationPanelAction implements Action<EmbeddableApiContext> {
```

Comment:
Any reason this action is in the Dashboard plugin and not in the Visualize plugin?

Reply:
I moved this to the visualizations plugin in [48bef80](https://github.com/elastic/kibana/pull/183764/commits/48bef80a677449e42f073b2107699a7a7368da25)